### PR TITLE
LPS-100844 Protecting timeout state calls with useIsMounted hook

### PR DESCRIPTION
**Pull request by 4lejandrito**

### CODE REVIEW

**Comment by 4lejandrito** [comment]
Hi @boton, @wincent, @p2kmgcl,

These are most of the changes we discussed yesterday. Things I didn't do:

* Use ` frontend-js-react-web`'s render function since it requires a componentId which I believe we don't have.
* Use context for the portletNamespace. We are still getting it from the store.

Regards

**Comment by wincent** [comment]
Thanks @4lejandrito.

FYI: Soon (https://issues.liferay.com/browse/LPS-100896) I plan to add some very common hooks to frontend-js-react-web, and a `useMounted` (or `useIsMounted`) hook will be one of them. (I'll update your code to use that when it goes in.)

Also, about `componentId`; Chema's stuff [went in yesterday](https://github.com/brianchandotcom/liferay-portal/pull/77801) — you can see there how it plucks `componentId` out of the passed-in data and generates a almost-certainly-unique id if you don't give it one.

**Comment by 4lejandrito** [comment]
@victorg1991 I forgot about you!

**Comment by 4lejandrito** [comment]
@p2kmgcl feel free to follow up on @wincent comment if you think is worth the effort. Otherwise just review my changes.

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/store/hooks/useIsMounted.es.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	const mountedRef = useRef(false);
+	const isMounted = useCallback(() => mountedRef.current, []);
+
+	useEffect(() => {
+		mountedRef.current = true;
+
+		return () => {
+			mountedRef.current = false;
+		};
+	});
```
Comment (4lejandrito) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/store/hooks/useIsMounted.es.js]: @wincent, @boton, @p2kmgcl, should we add `, []` to run this just on the first render?

Comment (wincent) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/store/hooks/useIsMounted.es.js]: Yeah, I think so. If you don't it's going to toggle back and forth (albeit harmlessly) on every render. Looking at 3 different implementations online, I see all of them pass `[]`.

Comment (boton) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/store/hooks/useIsMounted.es.js]: In the base implementation avoid the `[]` and I don't know why 🤔

Comment (4lejandrito) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/store/hooks/useIsMounted.es.js]: Done: https://github.com/p2kmgcl/liferay-portal/pull/448/commits/39d2f129f1da53d102102f1c556afca3cd6bcff5.

**Comment by p2kmgcl** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/78060

**Pull request closed without merging**